internal/pkg/runtime/engines/singularity/process_linux.go

```diff
@@ -145,8 +145,69 @@ func (engine *EngineOperations) checkExec() error {
 	return fmt.Errorf("no %s found inside container", args[0])
 }
 
+// startFuseDrivers starts the required FUSE drivers
+func startFuseDrivers(engine *EngineOperations) error {
+	for _, name := range engine.EngineConfig.GetPluginFuseMounts() {
+		var cfg struct {
+			Fuse struct {
+				DevFuseFd  int
+				MountPoint string
+				Program    []string
+			}
+		}
+
+		if err := engine.EngineConfig.GetPluginConfig(name, &cfg); err != nil {
+			return err
+		}
+
+		fh := os.NewFile(uintptr(cfg.Fuse.DevFuseFd), "fd-"+name)
+		fd := fh.Fd()
+		// The assumption is that the plugin prepared "Program"
+		// in such a way that it's missing the last parameter
+		// and that must correspond to /dev/fd/N
+		args := append(cfg.Fuse.Program, fmt.Sprintf("/dev/fd/%d", fd))
+		cmd := exec.Command(args[0], args[1:]...)
+
+		// Add the /dev/fuse file descriptor to the list of file
+		// descriptors to be passed to the new process.
+		//
+		// ExtraFiles is an array of *os.File, with the position
+		// of each entry determining the resulting file
+		// descriptor number. It must correspond to the file
+		// descriptor we have so that the above /dev/fd/%d
+		// works. Use fd-3 because we must skip over stdin,
+		// stdout, stderr; +1 because the second argument to
+		// make is the number of elements.
+		cmd.ExtraFiles = make([]*os.File, fd-3+1)
+		cmd.ExtraFiles[fd-3] = fh
+
+		if out, err := cmd.CombinedOutput(); err != nil {
+			sylog.Debugf("command: %v, output: %s\n", args, out)
+			return err
+		}
+	}
+
+	return nil
+}
+
+// preStartProcess does the final set up before starting the user's
+// process.
+func preStartProcess(engine *EngineOperations) error {
+	if err := startFuseDrivers(engine); err != nil {
+		return err
+	}
+
+	return nil
+}
+
 // StartProcess starts the process
 func (engine *EngineOperations) StartProcess(masterConn net.Conn) error {
+	// perform all the necessary "pre" operations, right before
+	// starting the user's process
+	if err := preStartProcess(engine); err != nil {
```

Comment:
As we already discussed, I have concerns about executing FUSE server inside container :
- it implies to compile FUSE server statically
- some FUSE server may need to access host filesystem or specific device which may not be made with few singularity flags like `--contain` enabled
- when container finish FUSE server can be killed without having time to flush pending data
- seccomp filter when enable could potentially affect FUSE server

Reply:
And as I have already said, that's a problem for the person implementing the plugin. That's precisely the reason why this is implemented like this.